rasa/utils/tensorflow/transformer.py

```diff
@@ -0,0 +1,478 @@
+from typing import List, Optional, Text, Tuple, Callable
+import tensorflow as tf
+import tensorflow_addons as tfa
+from tensorflow.python.keras.utils import tf_utils
+import numpy as np
+from rasa.utils.tensorflow.layers import DenseWithSparseWeights
+
+
+# from https://www.tensorflow.org/tutorials/text/transformer
+# and https://github.com/tensorflow/tensor2tensor
+class MultiHeadAttention(tf.keras.layers.Layer):
+    def __init__(
+        self,
+        d_model: int,
```

Comment:
Is it common to name the transformer size `d_model`? If not, can we rename it to `model_size` or `model_dimension`?

Reply:
it's just copy pasted from tf tutorial, I will rename

Reply:
in keras dense layers, it is called `units`